src/main/kotlin/org/sol4k/Transaction.kt

```diff
@@ -90,5 +100,93 @@ class Transaction(
+                    byteArray = byteArray.drop(PUBLIC_KEY_LENGTH).toByteArray()
+                    accountKeys.add(Base58.encode(account.toByteArray()))
+                }
+
+                val recentBlockhash = byteArray.slice(0 until PUBLIC_KEY_LENGTH).toByteArray()
+                byteArray = byteArray.drop(PUBLIC_KEY_LENGTH).toByteArray()
+
+                val instructionCount = Binary.decodeLength(byteArray)
+                byteArray = instructionCount.second
+                val instructions = mutableListOf<Instruction>()
+                for(i in 0 until instructionCount.first) {
+                    val programIdIndex = byteArray.first().toInt().also { byteArray = byteArray.drop(1).toByteArray() }
+                    val programId = accountKeys[programIdIndex]
+
+                    val accountCount = Binary.decodeLength(byteArray)
+                    byteArray = accountCount.second
+
+                    val accountIndices =
+                        byteArray.slice(0 until accountCount.first).toByteArray().toList().map(Byte::toInt)
+                    byteArray = byteArray.drop(accountCount.first).toByteArray()
+
+                    val dataLength = Binary.decodeLength(byteArray)
+                    byteArray = dataLength.second
+                    val dataSlice = byteArray.slice(0 until dataLength.first).toByteArray()
+                    byteArray = byteArray.drop(dataLength.first).toByteArray()
+                    instructions.add(
+                        BaseInstruction(
+                            programId = PublicKey(programId),
+                            data = dataSlice,
+                            keys = accountIndices.map { accountIdx ->
+                                AccountMeta(
+                                    publicKey = PublicKey(accountKeys[accountIdx]),
+                                    signer = accountIdx < numRequiredSignatures,
+                                    writable = accountIdx < numRequiredSignatures - numReadonlySignedAccounts ||
+                                            (accountIdx >= numRequiredSignatures &&
+                                                    accountIdx < accountKeys.count() - numReadonlyUnsignedAccounts)
+                                )
+                            }
+                        )
+                    )
+                }
+
+                // 3. construct Transaction
+                if(numRequiredSignatures <= 0) throw Exception("Feepayer does not exist")
+
+                Transaction(
+                    feePayer = PublicKey(accountKeys[0]),
+                    recentBlockhash = Base58.encode(recentBlockhash),
+                    instructions = instructions,
+                ).apply {
+                    signatures.forEach { signature ->
+                        this.addSignature(signature)
+                    }
+                }
+            } catch (e: Exception) {
```

Comment:
If the error is just re-thrown, the try-catch block carries no value. Let's unwrap it